File: packages/shared/src/types/device.ts
export interface Device {
    id: string;
    org_id: string;
    name: string;
    udid: string;
    platform: 'android' | 'ios';
    model?: string;
    manufacturer?: string;
    android_version?: string;
    screen_width?: number;
    screen_height?: number;
    status: 'online' | 'offline' | 'busy' | 'error';
    last_seen_at?: string;
}

File: apps/daemon/models/device.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum

class DeviceStatus(str, Enum):
    ONLINE = "ONLINE"
    OFFLINE = "OFFLINE"
    BUSY = "BUSY"

class Device(BaseModel):
    id: str
    udid: str
    name: str
    model: str
    os_version: str
    resolution: str
    status: DeviceStatus
    battery_level: Optional[int] = None
